--- kube-deploy/docker-multinode/common.sh.orig	2016-10-27 10:12:47.000000000 +0000
+++ kube-deploy/docker-multinode/common.sh	2016-12-02 15:20:30.000000000 +0000
@@ -36,7 +36,9 @@
     kube::log::fatal "Docker is not running on this machine!"
   fi
 
-  LATEST_STABLE_K8S_VERSION=$(curl -sSL "https://storage.googleapis.com/kubernetes-release/release/stable.txt")
+#  LATEST_STABLE_K8S_VERSION='v1.4.5-qiwi.5'
+  LATEST_STABLE_K8S_VERSION='v1.4.6'
+  #LATEST_STABLE_K8S_VERSION=$(curl -sSL "https://storage.googleapis.com/kubernetes-release/release/stable.txt")
   K8S_VERSION=${K8S_VERSION:-${LATEST_STABLE_K8S_VERSION}}
 
   CURRENT_PLATFORM=$(kube::helpers::host_platform)
@@ -48,10 +50,10 @@
     ETCD_VERSION=${ETCD_VERSION:-"3.0.4"}
   fi
 
-  FLANNEL_VERSION=${FLANNEL_VERSION:-"v0.6.1"}
+  FLANNEL_VERSION=${FLANNEL_VERSION:-"v0.6.2"}
   FLANNEL_IPMASQ=${FLANNEL_IPMASQ:-"true"}
-  FLANNEL_BACKEND=${FLANNEL_BACKEND:-"udp"}
-  FLANNEL_NETWORK=${FLANNEL_NETWORK:-"10.1.0.0/16"}
+  FLANNEL_BACKEND=${FLANNEL_BACKEND:-"vxlan"}
+  FLANNEL_NETWORK=${FLANNEL_NETWORK:-"172.17.0.0/16"}
 
   RESTART_POLICY=${RESTART_POLICY:-"unless-stopped"}
 
@@ -86,6 +88,13 @@
     ${KUBELET_MOUNT} \
     -v /var/log/containers:/var/log/containers:rw"
 
+  if [[ ${USE_CNI} == "true" ]]; then
+    KUBELET_MOUNTS="\
+      ${KUBELET_MOUNTS} \
+      -v /etc/cni/net.d:/etc/cni/net.d:rw \
+      -v /opt/cni/bin:/opt/cni/bin:rw"
+  fi
+
   # Paths
   FLANNEL_SUBNET_DIR=${FLANNEL_SUBNET_DIR:-/run/flannel}
 
@@ -131,10 +140,10 @@
     --restart=${RESTART_POLICY} \
     ${ETCD_NET_PARAM} \
     -v /var/lib/kubelet/etcd:/var/etcd \
-    gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+    dcr.qiwi.com/gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
     /usr/local/bin/etcd \
       --listen-client-urls=http://0.0.0.0:2379,http://0.0.0.0:4001 \
-      --advertise-client-urls=http://localhost:2379,http://localhost:4001 \
+      --advertise-client-urls=http://192.168.209.138:2379,http://192.168.209.138:4001 \
       --listen-peer-urls=http://0.0.0.0:2380 \
       --data-dir=/var/etcd/data
 
@@ -172,7 +181,7 @@
     --privileged \
     -v /dev/net:/dev/net \
     -v ${FLANNEL_SUBNET_DIR}:${FLANNEL_SUBNET_DIR} \
-    quay.io/coreos/flannel:${FLANNEL_VERSION}-${ARCH} \
+    dcr.qiwi.com/quay.io/coreos/flannel:${FLANNEL_VERSION}-${ARCH} \
     /opt/bin/flanneld \
       --etcd-endpoints=http://${MASTER_IP}:2379 \
       --ip-masq="${FLANNEL_IPMASQ}" \
@@ -208,7 +217,7 @@
     --restart=${RESTART_POLICY} \
     --name kube_kubelet_$(kube::helpers::small_sha) \
     ${KUBELET_MOUNTS} \
-    gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} \
+    dcr.qiwi.com/hyperkube-${ARCH}:${K8S_VERSION} \
     /hyperkube kubelet \
       --allow-privileged \
       --api-servers=http://localhost:8080 \
@@ -218,6 +227,7 @@
       ${CNI_ARGS} \
       ${CONTAINERIZED_FLAG} \
       --hostname-override=${IP_ADDRESS} \
+      --pod-infra-container-image=dcr.qiwi.com/gcr.io/google_containers/pause-amd64:3.0 \
       --v=2
 }
 
@@ -236,7 +246,7 @@
     --restart=${RESTART_POLICY} \
     --name kube_kubelet_$(kube::helpers::small_sha) \
     ${KUBELET_MOUNTS} \
-    gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} \
+    dcr.qiwi.com/hyperkube-${ARCH}:${K8S_VERSION} \
     /hyperkube kubelet \
       --allow-privileged \
       --api-servers=http://${MASTER_IP}:8080 \
@@ -245,6 +255,7 @@
       ${CNI_ARGS} \
       ${CONTAINERIZED_FLAG} \
       --hostname-override=${IP_ADDRESS} \
+      --pod-infra-container-image=dcr.qiwi.com/gcr.io/google_containers/pause-amd64:3.0 \
       --v=2
 }
 
@@ -257,7 +268,7 @@
     --privileged \
     --name kube_proxy_$(kube::helpers::small_sha) \
     --restart=${RESTART_POLICY} \
-    gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} \
+    dcr.qiwi.com/hyperkube-${ARCH}:${K8S_VERSION} \
     /hyperkube proxy \
         --master=http://${MASTER_IP}:8080 \
         --v=2
@@ -311,7 +322,7 @@
 
   # Remove cni0 bridge if K8S_VERSION = v1.4.0-alpha.2
   # With versions above that, it's not required. TODO: Remove this later
-  kube::multinode::delete_bridge cni0
+  #kube::multinode::delete_bridge cni0
 }
 
 kube::multinode::delete_bridge() {
@@ -388,7 +399,7 @@
 }
 
 kube::helpers::parse_version() {
-  local -r version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha)\\.(0|[1-9][0-9]*))?$"
+  local -r version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha|qiwi)\\.(0|[1-9][0-9]*))?$"
   local -r version="${1-}"
   [[ "${version}" =~ ${version_regex} ]] || {
     kube::log::fatal "Invalid release version: '${version}', must match regex ${version_regex}"
