--- ./kube-deploy/docker-multinode/common.sh.orig	2016-11-07 15:48:34.256521687 +0300
+++ ./kube-deploy/docker-multinode/common.sh	2016-12-15 15:07:18.856086110 +0300
@@ -36,7 +36,9 @@
     kube::log::fatal "Docker is not running on this machine!"
   fi
 
-  LATEST_STABLE_K8S_VERSION=$(curl -sSL "https://storage.googleapis.com/kubernetes-release/release/stable.txt")
+  LATEST_STABLE_K8S_VERSION='v1.5.1-qiwi.1'
+#  LATEST_STABLE_K8S_VERSION='v1.4.7'
+  #LATEST_STABLE_K8S_VERSION=$(curl -sSL "https://storage.googleapis.com/kubernetes-release/release/stable.txt")
   K8S_VERSION=${K8S_VERSION:-${LATEST_STABLE_K8S_VERSION}}
 
   CURRENT_PLATFORM=$(kube::helpers::host_platform)
@@ -86,6 +88,13 @@
     ${KUBELET_MOUNT} \
     -v /var/log/containers:/var/log/containers:rw"
 
+  if [[ ${USE_CNI} == "true" ]]; then
+    KUBELET_MOUNTS="\
+      ${KUBELET_MOUNTS} \
+      -v /etc/cni/net.d:/etc/cni/net.d:rw \
+      -v /opt/cni/bin:/opt/cni/bin:rw"
+  fi
+
   # Paths
   FLANNEL_SUBNET_DIR=${FLANNEL_SUBNET_DIR:-/run/flannel}
 
@@ -134,7 +143,7 @@
     gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
     /usr/local/bin/etcd \
       --listen-client-urls=http://0.0.0.0:2379,http://0.0.0.0:4001 \
-      --advertise-client-urls=http://localhost:2379,http://localhost:4001 \
+      --advertise-client-urls=http://192.168.209.138:2379,http://192.168.209.138:4001 \
       --listen-peer-urls=http://0.0.0.0:2380 \
       --data-dir=/var/etcd/data
 
@@ -208,7 +217,7 @@
     --restart=${RESTART_POLICY} \
     --name kube_kubelet_$(kube::helpers::small_sha) \
     ${KUBELET_MOUNTS} \
-    gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} \
+    dcr.qiwi.com/hyperkube-${ARCH}:${K8S_VERSION} \
     /hyperkube kubelet \
       --allow-privileged \
       --api-servers=http://localhost:8080 \
@@ -236,7 +245,7 @@
     --restart=${RESTART_POLICY} \
     --name kube_kubelet_$(kube::helpers::small_sha) \
     ${KUBELET_MOUNTS} \
-    gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} \
+    dcr.qiwi.com/hyperkube-${ARCH}:${K8S_VERSION} \
     /hyperkube kubelet \
       --allow-privileged \
       --api-servers=http://${MASTER_IP}:8080 \
@@ -257,7 +266,7 @@
     --privileged \
     --name kube_proxy_$(kube::helpers::small_sha) \
     --restart=${RESTART_POLICY} \
-    gcr.io/google_containers/hyperkube-${ARCH}:${K8S_VERSION} \
+    dcr.qiwi.com/hyperkube-${ARCH}:${K8S_VERSION} \
     /hyperkube proxy \
         --master=http://${MASTER_IP}:8080 \
         --v=2
@@ -311,7 +320,7 @@
 
   # Remove cni0 bridge if K8S_VERSION = v1.4.0-alpha.2
   # With versions above that, it's not required. TODO: Remove this later
-  kube::multinode::delete_bridge cni0
+#  kube::multinode::delete_bridge cni0
 }
 
 kube::multinode::delete_bridge() {
@@ -388,7 +397,7 @@
 }
 
 kube::helpers::parse_version() {
-  local -r version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha)\\.(0|[1-9][0-9]*))?$"
+  local -r version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha|qiwi)\\.(0|[1-9][0-9]*))?$"
   local -r version="${1-}"
   [[ "${version}" =~ ${version_regex} ]] || {
     kube::log::fatal "Invalid release version: '${version}', must match regex ${version_regex}"
