--- calico-containers/calico_node/startup.py.orig	2016-11-26 02:22:58.497514251 +0300
+++ calico-containers/calico_node/startup.py	2016-11-26 02:24:19.396834721 +0300
@@ -274,7 +274,7 @@
     # Get IP address of host, if none was specified
     if not ip:
         ips = get_host_ips(exclude=["^docker.*", "^cbr.*", "dummy.*",
-                                    "virbr.*", "lxcbr.*", "veth.*",
+                                    "virbr.*", "lxcbr.*", "veth.*", "^cni.*",
                                     "cali.*", "tunl.*", "flannel.*"])
         try:
             ip = str(ips.pop())
