--- ./kubernetes/cluster/addons/dns/skydns-rc.yaml.base.orig	2016-12-09 22:41:38.008036452 +0300
+++ ./kubernetes/cluster/addons/dns/skydns-rc.yaml.base	2016-12-09 22:41:10.076549435 +0300
@@ -48,7 +48,7 @@
     spec:
       containers:
       - name: kubedns
-        image: gcr.io/google_containers/kubedns-amd64:1.9
+        image: dcr.qiwi.com/gcr.io/google_containers/kubedns-amd64:1.9
         resources:
           # TODO: Set memory limits when we've profiled the container for large
           # clusters, then set request = limit to keep this container in
@@ -99,7 +99,7 @@
           name: metrics
           protocol: TCP
       - name: dnsmasq
-        image: gcr.io/google_containers/kube-dnsmasq-amd64:1.4
+        image: dcr.qiwi.com/gcr.io/google_containers/kube-dnsmasq-amd64:1.4
         livenessProbe:
           httpGet:
             path: /healthz-dnsmasq
@@ -127,7 +127,7 @@
             cpu: 150m
             memory: 10Mi
       - name: dnsmasq-metrics
-        image: gcr.io/google_containers/dnsmasq-metrics-amd64:1.0
+        image: dcr.qiwi.com/gcr.io/google_containers/dnsmasq-metrics-amd64:1.0
         livenessProbe:
           httpGet:
             path: /metrics
@@ -148,7 +148,7 @@
           requests:
             memory: 10Mi
       - name: healthz
-        image: gcr.io/google_containers/exechealthz-amd64:1.2
+        image: dcr.qiwi.com/gcr.io/google_containers/exechealthz-amd64:1.2
         resources:
           limits:
             memory: 50Mi
